src/uniprotkb/components/entry/similar-proteins/SimilarProteins.tsx

```diff
@@ -24,70 +26,70 @@ import {
   UniRefLiteAPIModel,
 } from '../../../../uniref/adapters/uniRefConverter';
 
+type IsoformsAndCluster = {
+  isoforms: string[];
+  cluster: UniRefLiteAPIModel;
+};
+
+type Mapping = Record<UniRefEntryType, IsoformsAndCluster[]>;
+
 type Props = {
   isoforms: { isoforms: string[] };
   primaryAccession: string;
 };
 
 const SimilarProteins = ({ isoforms, primaryAccession }: Props) => {
   const allAccessions = useMemo(
-    () => [primaryAccession, ...isoforms.isoforms],
+    () =>
+      [primaryAccession, ...isoforms.isoforms].filter(
+        // remove first isoform, it'll be the primary accession
+        (accession) => !accession.endsWith('-1')
+      ),
     [primaryAccession, isoforms]
   );
 
-  const searchUrl = getClustersForProteins(allAccessions);
-
-  // Get the clusters in which the canonical and isoforms are found
-  const { loading, data, error } = useDataApi<{
-    results: UniRefLiteAPIModel[];
-  }>(searchUrl);
-
-  const clusterData = useMemo(() => {
-    if (data) {
-      const { results } = data;
-      // Remove all items with only 1 member as it will be canonical/isoform
-
-      const filtered = results
-        .map((item) => ({
-          ...item,
-          members: item.members.filter(
-            (member) =>
-              !member.startsWith('UPI') && !allAccessions.includes(member)
-          ),
-        }))
-        .filter((item) => item.members.length > 1);
-
-      if (filtered.length === 0) {
-        return null;
+  const [mappingData, setMappingData] = useSafeState<Mapping | null>(null);
+  const [mappingLoading, setMappingLoading] = useSafeState(true);
+
+  useEffect(() => {
+    setMappingLoading(true);
+    const promises = allAccessions.map((accession) =>
+      fetchData<{
+        results: UniRefLiteAPIModel[];
+      }>(`${apiUrls.search(Namespace.uniref)}?query=(uniprot_id:${accession})`)
+    );
+    Promise.all(promises).then((responses) => {
+      const mapping: Mapping = {
+        UniRef100: [],
```

Comment:
Could do `UniRef100: {}` and use the `cluster.id` as the key to avoid needing a `find` later on

Reply:
I think I tried to optimise for render. Everything in this section should happen once, but the render might happen multiple times, so that's why I went for an array because this is what we'll use in the jsx section.
Not sure it would make a difference though, and so if it improves overall readability of the code I'd change it.
Will look tomorrow